### PR TITLE
New Haxe Operators Cheat Sheet

### PR DESCRIPTION
**Pull request by fzzr-**
IA: https://duck.co/ia/view/haxe_operators

### CODE REVIEW

**Comment by daxtheduck** [comment]
@fzzr- Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by fzzr-** [comment]
IA: https://duck.co/ia/view/haxe_operators

**Comment by fzzr-** [comment]
Hey, Travis!?
```
# Testing bittorrent............... 1 FAILURE
# 	 ⇒ FAIL: Invalid file name (bittorrent.json) for ID (bittorent_cheat_sheet)
# Testing haxe-ops................. PASS
```

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Haxe ops cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/haxe_operators_mobile.png)

File: share/goodie/cheat_sheets/json/haxe/haxe-ops.json [hunk 1]
```diff
@@ -0,0 +1,231 @@
+{
+  "id": "haxe_ops_cheat_sheet",
+  "name": "Haxe",
+  "version": "3.3.0",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/haxe/haxe-ops.json]: This should be part of the `name` we don't have a `version` property

Comment (fzzr-) [inline share/goodie/cheat_sheets/json/haxe/haxe-ops.json]: Yes. I can remove it, but I have some ideas:
1. I propose to add `version` field for version of tool;
2. I propose to add `locale` field for future localization.

Comment (moollaza) [inline share/goodie/cheat_sheets/json/haxe/haxe-ops.json]: > 1. I propose to add `version` field for version of tool;

Right now I think we only have 1 version for more software covered. I think there might be value in an explicit, `version:` property, but I think there's a larger issue of necessity, triggering and organization.

- Do we need cheat sheets for older versions?
- Do we require explicit triggers for each version? Or should we allow them to change the version on the page, and updated the content?
- How do we organize cheat sheets that cover different versions of software? In sub-folders? Or all at the same level?

I think this also ties in to the discussion we have about handling cheat sheets for different OS's. IE having a Linux/Windows/Mac version where necessary, showing the correct one based on the user's system and allowing them to switch between them from the result page (so you don't need to make your query verbose)

Comment (moollaza) [inline share/goodie/cheat_sheets/json/haxe/haxe-ops.json]: > 2. I propose to add `locale` field for future localization.

Can you expand on this? I don't understand the use case

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Haxe ops cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/haxe_operators_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Haxe ops cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/haxe_operators_mobile.png)

**Comment by GuiltyDolphin** [comment]
@fzzr- Thanks! :dolphin:

**Comment by GuiltyDolphin** [comment]
@moollaza @zekiel Can we get the IA page renamed to `haxe_operators_cheat_sheet`?

**Comment by zekiel** [comment]
@fzzr- @GuiltyDolphin updated! The name is now Hax Operators Cheat Sheet and ID is set to `haxe_operators_cheat_sheet`

**Comment by GuiltyDolphin** [comment]
@fzzr- Thanks! :dolphin:

**Comment by daxtheduck** [comment]
@fzzr- this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by fzzr-** [comment]
Thx!

**Pull request merged**